internal/codegen/shared_state.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2024 stencil contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codegen
+
+import (
+	"path"
+	"sort"
+
+	"github.com/mitchellh/hashstructure/v2"
+	"github.com/puzpuzpuz/xsync/v3"
+)
+
+// hashModuleHookValue hashes the module hook value using the
+// hashstructure library. If the hashing fails, it returns 0.
+func hashModuleHookValue(m any) uint64 {
+	hash, err := hashstructure.Hash(m, hashstructure.FormatV2, nil)
+	if err != nil {
+		hash = 0
```

Comment:
you may want to panic here until we'd understand what this failure case looks like.  otherwise you won't get any passes...

Reply:
That's for module hooks only, which we use to sort. If we fail to sort we use `0` because it's _something_ to sort w/ rather than nothing.

Reply:
i get that, but still, if it's really unexpected, let's blow up and sort it out later if we ever hit it

Reply:
hmmm fair enough. Let me look at the cases where it'd fail to begin with